subprojects/logging/src/main/java/org/gradle/internal/logging/console/ProgressOperation.java

```diff
@@ -65,7 +65,21 @@ public ProgressOperation getParent() {
         return parent;
     }
 
-    public Set<ProgressOperation> getChildren() {
-        return children;
+    public boolean addChild(ProgressOperation operation) {
+        if (children == null) {
```

Comment:
❌ I assume there's no thread safety concerns?  It's possible for an operation to be lost if multiple threads can add operations.